File: 01-twoSumUp/two-sum-up.ts
// O(nlog(n)) | O(1) space
export function twoNumberSum(array: number[], targetSum: number): number[] {
  array.sort((a, b) => a - b);

  let left: number = 0;
  let right: number = array.length - 1;

  while (left < right) {
    const currentSum: number = array[left] + array[right];

    if (currentSum === targetSum) {
      return [array[left], array[right]];
    } else if (currentSum < targetSum) {
      left += 1;
    } else if (currentSum > targetSum) {
      right -= 1;
    }
  }

  return [];
}

File: 01-twoSumUp/two-sum-up.swift
// O(nlog(n)) | O(1) space
func twoNumberSum(array: [Int], targetSum: Int) -> [Int] {
  let array = array.sorted()
	var leftPointer: Int = 0
	var rightPointer: Int = array.count - 1

	while (leftPointer < rightPointer) {
		let currentSum = array[leftPointer] + array[rightPointer]

		if currentSum == targetSum {
			return [array[leftPointer], array[rightPointer]]
		} else if currentSum < targetSum {
			leftPointer += 1
		} else if currentSum > targetSum {
			rightPointer -= 1
		}
	}

	return []
}
